### PR TITLE
edm::View<reco::Track> instead of reco::TrackCollection in DQM/TrackingMonitor

### PR DESCRIPTION
**Pull request by mtosi**
in order to be able to monitor reco::GsfTrack by using the already well advanced DQM/TrackMonitor
the reco::TrackCollection has been changed into edm::View<reco::Track>

code changes have been tested by running
`runTheMatrix -l limited`
and all workflows seem fine

in addition, we have --finally-- (filled) distributions for hltGsfTracks
hltgsftracks

@rovere @VinInn @gpasztor @dmitrijus @vanbesien

### CODE REVIEW

**Comment by mtosi** [comment]
it would be great if this PR could be integrated in 900pre3, thanks

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by dmitrijus** [comment]
+1

File: DQM/TrackingMonitor/src/TrackingMonitor.cc [hunk 1]
```diff
@@ -707,15 +708,15 @@ void TrackingMonitor::analyze(const edm::Event& iEvent, const edm::EventSetup& i
...
+      const edm::View<reco::Track>& trackCollection = *trackHandle;
       // calculate the mean # rechits and layers
       int totalRecHits = 0, totalLayers = 0;
 
       theTrackAnalyzer->setNumberOfGoodVertices(iEvent);
       theTrackAnalyzer->setBX(iEvent);
       theTrackAnalyzer->setLumi(iEvent,iSetup);
-      for (reco::TrackCollection::const_iterator track = trackCollection.begin();
-	   track!=trackCollection.end(); ++track) {
+      for ( edm::View<reco::Track>::const_iterator track = trackHandle->begin();
```
Comment (davidlange6) [inline DQM/TrackingMonitor/src/TrackingMonitor.cc]: @mtosi - please iterate over the track collection not the handle to it.

**Comment by mtosi** [comment]
thanks for your feedback
I'm currently on the train to turin, I'll update this PR asap
when is the deadline for 900pre3 ?
in case it is tomorrow morning, may I ask you to merge this PR --as it is-- 
and let me add a new PR for handling this comment (I'll do it) to be merged in pre4
please ?

thanks

**Comment by mtosi** [comment]
@davidlange6

**Comment by davidlange6** [comment]
Hi @mtosi - pre3 won't happen tomorrow as far as I can see

**Comment by mtosi** [comment]
hope it is fine, now
thanks for your help
@davidlange6

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**